src/main/clojure/conexp/fca/factorization.clj

```diff
@@ -0,0 +1,927 @@
+(ns conexp.fca.factorization
+    (:require [conexp.base :refer :all]
+              [conexp.fca.fast :refer [to-binary-matrix]]
+              [conexp.fca.contexts :refer :all]))
+
+(defprotocol factorization-protocol
+  (context [this] "Returns the context that is generated after using the algorithm algo k-times on it")
+  (object-factor [this] "Returns the object-factor of the context generated by the algorithm")
+  (attribute-factor [this] "Returns the attribute-factor of the context generated by the algorithm"))
+
+(defrecord factorization-record [context object-factor attribute-factor]
+  factorization-protocol
+  (context [this] context)
+  (object-factor [this] object-factor)
+  (attribute-factor [this] attribute-factor))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; helper functions
+
+(defn make-matrix-from-concept
```

Comment:
Cant this be done with the map function?

Reply:
This is the best I came up with :D